src/main/java/com/prgrms/vouchermanager/io/FileIO.java

```diff
@@ -0,0 +1,93 @@
+package com.prgrms.vouchermanager.io;
+
+import com.prgrms.vouchermanager.domain.voucher.FixedAmountVoucher;
+import com.prgrms.vouchermanager.domain.voucher.PercentAmountVoucher;
+import com.prgrms.vouchermanager.domain.voucher.Voucher;
+import com.prgrms.vouchermanager.exception.FileIOException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.*;
+import java.util.Map;
+import java.util.UUID;
+
+@Slf4j
+public class FileIO {
+    private final BufferedReader bf;
+    private final String filePath;
+
+    public FileIO(String filePath) {
+        this.filePath = filePath;
+        try {
+            bf = new BufferedReader(new FileReader(filePath));
+        } catch (FileNotFoundException e) {
+            FileIOException ex = new FileIOException(this);
+            log.error(ex.getMessage());
+            throw ex;
+        }
+    }
+
+    public void updateFile(Map<UUID, Voucher> vouchers) {
+        try {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(filePath));
+            vouchers.forEach((key, voucher) -> {
+                UUID id = voucher.getId();
+                String type = voucher instanceof FixedAmountVoucher ? "fixed" : "percent";
+                long discount = voucher.getDiscount();
+
+                try {
+                    bw.write(id + "," + type + "," + discount + "\n");
+                } catch (IOException e) {
+                    RuntimeException ex = new FileIOException(bw);
+                    log.error(ex.getMessage());
+                    throw ex;
+                }
+            });
+            bw.close();
+        } catch (IOException e) {
+            RuntimeException ex = new FileIOException(this);
+            log.error(ex.getMessage());
+            throw ex;
+        }
+    }
+
+    public void fileToCustomerMap(Map<UUID, com.prgrms.vouchermanager.domain.customer.Customer> customerMap) {
+        String line = "";
+
+        while(true) {
+            try {
+                if ((line = bf.readLine()) == null) break;
+            } catch (IOException e) {
+                RuntimeException ex = new FileIOException(bf);
+                log.error(ex.getMessage());
+                throw ex;
+            }
```

Comment:
```suggestion
        try (BufferedReader bf = new BufferedReader(new FileReader(fileName))) {
            String line;
            while ((line = bf.readLine()) != null) {
                // 스플릿 후 map에 추가 하는 메서드 작성하여 책임 분리
            }
        } catch (IOException e) {
            RuntimeException ex = new FileIOException(e);
            log.error(ex.getMessage(), e);
            throw ex;
        }
```

생성자에서 BufferedReader 를 초기화 하고 사용하는 것보다 메서드 마다 사용하되 
try with resources 방식으로 자원을 닫는 코드가 있는게 더 안정적이지 않을까 싶습니다.